axe_selenium_python/tests/test_axe.py

```diff
@@ -5,33 +5,44 @@
 from os import path
 
 import pytest
+from selenium import webdriver
 
+from ..axe import Axe
 
-@pytest.mark.nondestructive
-def test_execute(axe):
-    """Run axe against base_url and verify JSON output."""
-    axe.inject()
-    data = axe.execute()
-    assert data is not None, data
+_DEFAULT_TEST_FILE = path.join(path.dirname(__file__), 'test_page.html')
 
 
 @pytest.mark.nondestructive
-def test_report(axe):
-    """Test that report exists."""
+def test_run_axe_sample_page_firefox():
+    """Run axe against sample page and verify JSON output is as expected."""
+    driver = webdriver.Firefox()
+    driver.get('file://' + _DEFAULT_TEST_FILE)
+    print(_DEFAULT_TEST_FILE)
+    axe = Axe(driver)
+
     axe.inject()
-    results = axe.execute()
-    violations = results["violations"]
-    report = axe.report(violations)
-    assert report is not None, report
+    data = axe.execute()
+    driver.close()
+
+    assert len(data["inapplicable"]) == 46
+    assert len(data["incomplete"]) == 0
+    assert len(data["passes"]) == 7
+    assert len(data["violations"]) == 8
 
 
 @pytest.mark.nondestructive
-def test_write_results(base_url, axe):
-    """Assert that write results method creates a non-empty file."""
+@pytest.mark.xfail(reason="known issue in chrome: https://github.com/mozilla-services/axe-selenium-python/issues/118")
```

Comment:
Using the suggested parameterisation, you can mark a single parameter as expected to fail:

```python
@pytest.fixture(params=(
    'Firefox',
    pytest.mark.xfail('Chrome', reason='known issue in chrome...')))
```